analysis/dataset_clean/fn-modify_dummy.R

```diff
@@ -0,0 +1,385 @@
+                    ~if_else(
+                      modify_mixed_vax,
+                      mixed_vax_threshold-2,
+                      .x)))%>%
+      mutate(across(vax_date_covid_2,
+                    ~if_else(
+                      modify_mixed_vax,
+                      mixed_vax_threshold-2,
+                      .x))) 
+    
+    
+      # Inclusion criteria: Index date is before cohort end date
+
+  } else if (cohort == "unvax") { # Modifying unvax-specific variables
+
+    df <- df %>%
+
+    ## JCVI distribution
+
+      mutate(vax_cat_jcvi_group = sample(
+      x       = c("12","11","10",
+                  "09","08","07",
+                  "06","05","04",
+                  "03","02","01","99"), # 8.25 for each group, 1.6% for missing
+      size    = nrow(.),
+      replace = TRUE,
+      prob    = c(0.082,0.082,0.082,0.082,0.082,0.082,0.082,0.082,0.082,0.082,0.082,0.082, 0.016)))
+
+  }
+
+  ## Modifying variables across cohorts
+
+  ## inex_bin_alive ----------------------------------------------------------
+
+  df <- df %>%
+
+    ## Alive on the index date
+    mutate(inex_bin_alive = rbernoulli(nrow(.), p = 0.99)) %>% 
+
+    # could also modify cens_date_death to match
+
+    ## Registered for a minimum of 6 months prior to index date
+    mutate(inex_bin_6m_reg = rbernoulli(nrow(.), p = 0.99)) %>% 
+
+    ## Age distribution
+    mutate(cov_num_age = sample(
+                        c(sample(  1:17,          round(nrow(.)*0.02),  replace=TRUE), # Number <18
+                          sample(110:120,         round(nrow(.)*0.02),  replace=TRUE), # Number >110
+                          sample( 18:110, nrow(.)-round(nrow(.)*0.02)*2,replace=TRUE)))) %>% # Remainder
+
+    ## Recalculate birth year based on new age
+    mutate(qa_num_birth_year = as.numeric(format(as.Date(index_date), "%Y"))-cov_num_age) %>%
+
+    ## Sex
+    mutate(cov_cat_sex = sample(
+                  x       = c("female",
+                              "male",
+                              "intersex",
+                              "unknown"), # %49% Female, 49% Male, 1% Intersex, 1% missing
+                  size    = nrow(.),
+                  replace = TRUE,
+                  prob    = c(0.49, 0.49, 0.01, 0.01))) %>%
+
+    ## Region
+    mutate(strat_cat_region = sample(
+                  x       = c("East",
+                              "East Midlands",
+                              "London",
+                              "North East",
+                              "North West",
+                              "South East",
+                              "South West",
+                              "West Midlands",
+                              "Yorkshire and The Humber",
+                              ""), # 11% for each area, %1 Missing
+                  size    = nrow(.),
+                  replace = TRUE,
+                  prob    = c(0.11, 0.11, 0.11, 0.11, 0.11, 0.11, 0.11, 0.11, 0.11, 0.01))) %>%
```

Comment:
Use `rep()` as demonstrated above for JCVI group.

Reply:
I think it is fine - it is not realistic, but variation is actually what is important here so that we expose the code to as many combinations of things as possible.